guidance/models/llama_cpp/_llama_cpp.py

```diff
@@ -89,9 +89,28 @@ def __init__(self, model_obj, chat_template=None):
         )
 
     def encode(self, byte_string: bytes) -> Sequence[int]:
+        # Some models (e.g. Llama3) can produce tokens which are only partial
+        # UTF-8 codepoints. The underlying tokenizer can't cope with these
+        # and has a tendency to segfault.
+        # To address this, shorten the bytes sent to the tokenizer to a valid
+        # UTF-8 value
+        # I hope this will not bite us with some subtle other bug in future
```

Comment:
I'm not very optimistic about this. This 'fix' almost certainly breaks `recode()` when it kicks in. And fundamentally, these bytes came from the LLM, so why on earth is its tokeniser refusing to have anything to do with them?

Reply:
@knilink had found:
`printf '\xe6\xad' | ./llama-tokenize -m ./Meta-Llama-3-8B-Instruct.Q8_0.gguf --stdin`

I have filed a `llama-cpp-python` issue (based on @knilink 's investigation) directly on the HF repo whence I'm grabbing the GGUF

Reply:
And filed the LlamaCpp bug:
https://github.com/ggerganov/llama.cpp/issues/8691